nautobot/core/jobs/bulk_actions.py

```diff
@@ -0,0 +1,156 @@
+from django.contrib.contenttypes.models import ContentType
+from django.core.exceptions import (
+    FieldDoesNotExist,
+    ObjectDoesNotExist,
+    PermissionDenied,
+    ValidationError,
+)
+from django.db.models import ManyToManyField
+
+from nautobot.core.forms.utils import restrict_form_fields
+from nautobot.core.utils.lookup import get_filterset_for_model, get_form_for_model
+from nautobot.extras.context_managers import deferred_change_logging_for_bulk_operation
+from nautobot.extras.jobs import (
+    BooleanVar,
+    Job,
+    JSONVar,
+    ObjectVar,
+    RunJobTaskFailed,
+)
+from nautobot.extras.utils import remove_prefix_from_cf_key
+
+
+class BulkEditObjects(Job):
+    """System Job to bulk Edit objects."""
+
+    content_type = ObjectVar(
+        model=ContentType,
+        description="Type of objects to update",
+    )
+    post_data = JSONVar(description="Data to update with i.e request.POST")
+    edit_all = BooleanVar(description="Bulk Edit all object / all filtered objects", required=False)
+    filter_query_params = JSONVar(label="Filter Query Params", required=False)
+
+    class Meta:
+        name = "Bulk Edit Objects"
+        description = "Bulk edit objects."
+        has_sensitive_variables = False
+        soft_time_limit = 1800
+        time_limit = 2000
+
+    def _update_objects(self, model, form, filter_query_params, edit_all, nullified_fields):
+        with deferred_change_logging_for_bulk_operation():
+            updated_objects_pk = []
+            filterset_cls = get_filterset_for_model(model)
+            if edit_all:
+                if filterset_cls:
+                    queryset = filterset_cls(filter_query_params).qs.restrict(self.user, "change")
```

Comment:
Should we log and/or raise an error if there are `filter_query_params` but we can't find the `filterset_cls`?